### PR TITLE
copy header text in channel info implemented

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR adds the feature to copy channel header text.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24613

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [X] Has UI changes
- [X] Includes text changes and localization file updates
- [X] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on:  Android 12

#### Release Note

```release-note
Implemented copy channel header text functionality.
```

### CODE REVIEW

**Comment by tanmaythole** [comment]
Hi @larkox,
Can you please look into it?

File: app/screens/channel_info/extra/extra.tsx [hunk 1]
```diff
@@ -81,6 +104,79 @@ const Extra = ({channelId, createdAt, createdBy, customStatus, header, isCustomS
...
 
+    const onCopy = useCallback(async (text: string, isLink?: boolean) => {
+        Clipboard.setString(text);
+        await dismissBottomSheet();
+        if ((Platform.OS === OS_VERSION.ANDROID && Number(Platform.Version) < ANDROID_33) || Platform.OS === OS_VERSION.IOS) {
+            showSnackBar({
+                barType: isLink ? SNACK_BAR_TYPE.LINK_COPIED : SNACK_BAR_TYPE.TEXT_COPIED,
+            });
+        }
+    }, []);
```
Comment (larkox) [inline app/screens/channel_info/extra/extra.tsx:115]: I may be missing something, but this function does not require anything from within the component, right? If that is so, we can move it out of the component and avoid creating it on every render.

File: app/screens/channel_info/extra/extra.tsx [hunk 2]
```diff
@@ -131,25 +227,32 @@ const Extra = ({channelId, createdAt, createdBy, customStatus, header, isCustomS
...
-                    layoutWidth={100}
-                    theme={theme}
-                    imagesMetadata={headerMetadata}
-                    value={header}
+                    testID={headerTestId}
                 />
+                <TouchableWithFeedback
+                    type='opacity'
+                    activeOpacity={0.8}
+                    onLongPress={() => handleLongPress()}
```
Comment (larkox) [inline app/screens/channel_info/extra/extra.tsx:235]: We are passing here a new function on every render. I think I understand why you don't do just `onLongPress={handleLongPress}` since onLongPress will send an event, and will mess with the logic of `handleLongPress`. If that is the case, maybe another `useCallback` would work here.
```
const touchableHandleLongPress = useCallback(() => handleLongPress(), [handleLongPress]);
```

File: app/components/markdown/markdown_link/markdown_link.tsx [hunk 3]
```diff
@@ -109,14 +110,19 @@ const MarkdownLink = ({children, experimentalNormalizeMarkdownLinks, href, siteU
...
+
             const renderContent = () => {
                 return (
                     <View
                         testID='at_mention.bottom_sheet'
                         style={style.bottomSheet}
                     >
                         <SlideUpPanelItem
-                            leftIcon='content-copy'
+                            leftIcon='link-variant'
```
Comment (larkox) [inline app/components/markdown/markdown_link/markdown_link.tsx:125]: Why this change?

**Review by larkox — APPROVED** [review]
Just one question and two minor improvements, but looks good to me.

And thank you for your patience. Sometimes the PRs slip through the cracks... :sweat_smile:

File: app/screens/channel_info/extra/extra.tsx (hunk 1 above)
Comment (tanmaythole) [reply app/screens/channel_info/extra/extra.tsx]: yeah, you're right. we can write it outside component

File: app/components/markdown/markdown_link/markdown_link.tsx (hunk 3 above)
Comment (tanmaythole) [reply app/components/markdown/markdown_link/markdown_link.tsx]: oops, It's my mistake

**Review by harshilsharma63 — APPROVED** [review]
Awesome! Looks good!

**Comment by enahum** [comment]
@harshilsharma63 @larkox who is the QA assigned to review this one?

**Comment by yasserfaraazkhan** [comment]
- If we long press on , Text/text that is hyperlink, it should `Copy header text` and `copy URL`
when long press on text

- If we long press on , some other part of header as highlight, it shows `Copy header text`
when long pressed on non text area

In both the option, the whole header is copied as expected and also can copy just the URL.

image

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by larkox** [comment]
Thank you for the contribution!
